### PR TITLE
Create sophos-endpoint-security

### PR DESCRIPTION
**Pull request by mikiodehartj1**
Initial barebones build. Will updated further once integration goes into beta

### Alerts Rule Packs
- [ ] Does a related alert already exist, and should it be tweaked or added to instead?
- [ ] Test each query to make sure it works
- [ ] Look for hardcoded variables/parameter values in the query
- [ ] Consider Severity for Alerts
- [ ] Spellcheck
- [ ] Use all caps for J1QL keywords and relationship classes
- [ ] Upload the alerts rule pack JSON into JupiterOne to validate

### CODE REVIEW

File: rule-packs/sophos-endpoint-security [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        "name": "query0",
+        "query": "FIND sophos_endpoint with tamperProtectionEnabled != true ",
+        "version": "v1"
+      }
+    ]
+    "alertLevel": "MEDIUM"
+  },
+  {
+    "name": "sophos-misconfigured-endpoints-4",
+    "description": "This query is looking for endpoints with an unacceptable lockdown status. Any status locked, notInstalled, unavailable,uninstalled is considered unacceptable",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:41]: unexpected vs. unacceptable

File: rule-packs/sophos-endpoint-security [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        "name": "query0",
+        "query": "FIND sophos_endpoint with lockdownStatus != ("installing" or "registering" or "starting" or "stopping" or "unlocked")",
+        "version": "v1"
+      }
+    ]
+    "alertLevel": "MEDIUM"
+  },
+  {
+    "name": "sophos-misconfigured-endpoints-5",
+    "description": "This query is looking for endpoints with an unacceptable update status. This includes any endpoint with a status of notInstalled.",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:53]: unexpected

File: rule-packs/sophos-endpoint-security [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        "name": "query0",
+        "query": "FIND sophos_endpoint with updateStatus != ("upToDate" or "updating" or "rebootRequired")",
+        "version": "v1"
+      }
+    ]
+    "alertLevel": "MEDIUM"
+  },
+  {
+    "name": "sophos-misconfigured-endpoints-6",
+    "description": "This query will look for endpoints with misconfigured ports. This will help in determining potential rogue devices or devices of risk.",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:65]: "with unusual listening ports enabled.."

File: rule-packs/sophos-endpoint-security [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+      {
+        "name": "query0",
+        "query": "FIND sophos_device with portsEnabled != "443"",
+        "version": "v1"
+      }
+    ]
+    "alertLevel": "MEDIUM"
+  },
+  {
+    "name": "domain-expires-in-30-days",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:76]: does this rule belong in the Sophos pack? looks unrelated

File: rule-packs/sophos-endpoint-security [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    ]
+    "alertLevel": "MEDIUM"
+  },
+  {
+    "name": "sophos-misconfigured-endpoints-9",
+    "description": "This query will return endpoints that have an unencrypted status.",
+    "queries": [
+      {
+        "name": "query0",
+        "query": "FIND sophos_enpoint with encryption != true",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:117]: sophos_endpoint

File: rule-packs/sophos-endpoint-security [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    ]
+    "alertLevel": "MEDIUM"
+  },
+]
+#end-sophos-endpoint-misconfigurations
+#start-sophos-user-app-misconfigurations
+[
+  {
+    "name": "sophos-misconfigured-endpoints-1",
+    "description": "This will query users that are potentially inactive but haven't been deactivated.",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:129]: 'query Sophos platform users..'

File: rule-packs/sophos-endpoint-security [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    ]
+    "alertLevel": "MEDIUM"
+  },
+  {
+    "name": "sophos-misconfigured-endpoints-2",
+    "description": "This will use the permissionSet fields to determine service/user permissions to ensure that separation of duties is established.",
+    "queries": [
+      {
+        "name": "query0",
+        "query": "FIND sophos_user that has a updatedAt > 30 days",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:145]: duplicate query

File: rule-packs/sophos-endpoint-security [hunk 8]
```diff
@@ -0,0 +1,166 @@
+#start-sophos-endpoint-misconfigurations
+[
+  {
+    "name": "sophos-misconfigured-endpoints-1",
+    "description": "This will query for endpoints that do not have sophos edr installed.",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:5]: 'that do not have the Sophos EDR Agent installed."

File: rule-packs/sophos-endpoint-security [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    ]
+    "alertLevel": "MEDIUM"
+  },
+  {
+    "name": "sophos-misconfigured-endpoints-2",
+    "description": "This will query for endpoints that are missing a relationship with a user. This will help determine potential rogue devices",
+    "queries": [
+      {
+        "name": "query0",
+        "query": "Find sophos_sensor that !HAS User THAT IS Person THAT IS okta_user|azure_user|google_user",
```
Comment (SeaBlooms) [inline rule-packs/sophos-endpoint-security:21]: is it 'sophos_device', 'sophos_endpoint' or 'sophos_sensor'? I think a mix of these is used in the rules and all need to match

**Review by SeaBlooms — CHANGES_REQUESTED** [review]
some updates needed, great start!

File: rule-packs/sophos-endpoint-security (hunk 9 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security:21]: sophos endpoint is the sensor
sophos device is the device

File: rule-packs/sophos-endpoint-security (hunk 8 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security]: Adjusting

File: rule-packs/sophos-endpoint-security (hunk 2 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security]: adjusted

File: rule-packs/sophos-endpoint-security (hunk 1 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security]: adjusted

File: rule-packs/sophos-endpoint-security (hunk 3 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security]: adjustedf

File: rule-packs/sophos-endpoint-security (hunk 4 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security]: deleted

File: rule-packs/sophos-endpoint-security (hunk 5 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security]: adjusted

File: rule-packs/sophos-endpoint-security (hunk 6 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security]: fixed

File: rule-packs/sophos-endpoint-security (hunk 7 above)
Comment (mikiodehartj1) [reply rule-packs/sophos-endpoint-security]: forgot to add in the query - fixed.

**Review by mikiodehartj1 — COMMENTED** [review]
corrections made per the comments

**Review by mikiodehartj1 — COMMENTED** [review]
adjusting the index.js

**Review by mikiodehartj1 — COMMENTED** [review]
change syntax issue

**Review by mikiodehartj1 — COMMENTED** [review]
a few syntax updates.

**Review by SeaBlooms — APPROVED** [review]
